controller.go

```diff
@@ -63,17 +72,18 @@ func (c *K8SController) mainloop(item interface{}) {
 		c.OnUpdate(obj)
 	case ControllerEventTypeDelete:
 		c.OnDelete(obj)
+	default:
+		log.Warn().Msgf("no event handler for '%s', event type '%s'", event.Key, event.Type)
```

Comment:
I'm actually not sure this is a good default.  I actually almost think a panic is proper here because otherwise if kubernetes ever creates a new event type besides Create, Update and Delete we will only catch it if we are watching the logs.  A panic would cause the operation to crash and we'd know about it way sooner.

If this was something we ran in our own infrastructure i'd say log a sentry issue and don't crash but because it doesn't logging warning feels like we'll never be notified of this unless someone stumbles upon the log line.

Reply:
Closing this conversation since we discussed we will leave this as warn since it's better than nothing even though it's very likely someone will ever see it. It's better than `panic`ing because we don't want a container that runs 24/7 to panic unless something is seriously broken to the point of causing data corruption or loss.